apps/services/auth-public-api/test/fixtures/delegation.fixture.ts

```diff
@@ -0,0 +1,129 @@
+  | 'identityResourceName'
+  | 'delegationId'
+>
+
+export type CreateDelegation = Pick<
+  Delegation,
+  'id' | 'fromNationalId' | 'fromDisplayName' | 'toNationalId' | 'toName'
+> & { delegationScopes: CreateDelegationScope[] }
+
+const createRandomNationalId = (): string => {
+  return faker.datatype
+    .number({
+      min: 10000000000,
+      max: 9999999999,
+    })
+    .toString()
+}
+
+/**
+ * Private helper to create a DelegationScope with random values.
+ * @param delegationId Id of the delegation the scopes belongs to.
+ * @returns
+ */
+const createRandomDelegationScope = (
+  delegationId: string,
+): CreateDelegationScope => {
+  return {
+    id: faker.datatype.uuid(),
+    validFrom: faker.datatype.datetime(),
+    validTo: faker.datatype.datetime(),
+    scopeName: faker.random.word(),
+    delegationId,
+  }
+}
+
+/**
+ * Private helper to create a Delegation with random values.
+ * @returns
+ */
+const createRandomDelegation = (): CreateDelegation => {
+  const id = faker.datatype.uuid()
+
+  return {
+    id,
+    fromNationalId: createRandomNationalId(),
+    fromDisplayName: faker.random.word(),
+    toNationalId: createRandomNationalId(),
+    toName: faker.random.word(),
+    delegationScopes: [createRandomDelegationScope(id)],
+  }
+}
+
+/**
+ * Creates a DelegationScope fixture to be used with Delegation for testing.
+ * @param delegationId Id of the delegation that the scope belongs to.
+ * @param scopeName Name of the scope.
+ * @param today A date the fixture uses as the current day to set valid or expired scopes.
+ *              The test suite should use useFakeTimers and provide this value for consistent testing.
+ *              Defaults to current datetime, but note that it could lead to undesired test results.
+ * @param expired Flag to indicate if the delgation should be expired. Defaults to false.
+ * @returns
+ */
+export const createDelegationScope = (
+  delegationId: string,
+  scopeName: string,
+  today: Date,
+  expired: boolean,
+  future: boolean,
+): CreateDelegationScope => {
+  const fallback = createRandomDelegationScope(delegationId)
+
+  return {
+    ...fallback,
+    scopeName,
+    validFrom: startOfDay(addDays(today, expired ? -2 : future ? 1 : 0)),
+    validTo: startOfDay(addDays(today, expired ? -1 : future ? 2 : 1)),
+  }
+}
+
+/**
+ * Creates a Delegation fixture to be used for testing.
+ * @param fromNationalId NationalId of the user granting the delegation
+ * @param toNationalId NationalId of the user receiving the delegation
+ * @param scopes Scopes the delegation grants access to
+ * @param today A date the fixture uses as the current day to set valid or expired scopes.
+ *              The test suite should use useFakeTimers and provide this value for consistent testing.
+ *              Defaults to current datetime, but note that it could lead to undesired test results.
+ * @param expired Flag to indicate if the delgation should be expired. Defaults to false.
```

Comment:
nit-picky, but comments should be treated like second-class citizens. They tend to get outdated once developers update the code.
Code should be able to explain itself with good names, thus reduces the need for comments which then only provide extra noise.

e.g. if a developers decides to rename or remove a parameter, there's nothing that forces him/her to remove/rename the comment as well, thus the comments become outdated.

Reply:
Valid point. Simplified the comment. I still find it valuable to get further details on autocomplete pop up.

Reply:
Yeah I see your point ... `vim` user here :D.